### ISSUE TITLE
NetworkWithInputEncoding is always initialized with the same random values

### ISSUE THREAD

**Issue by mlamarre** [NVlabs/tiny-cuda-nn#512]
Pytorch 2.8.0, CUDA 12.8

tiny-cuda-nn installed like this (Aug. 11th)
`uv pip install --no-build-isolation git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch`

Toolchain:
Visual Studio 2022, ninja.

`tcnn.NetworkWithInputEncoding` is always initialized with the same random values.

I would expect a different initialization at each call. 

Test code:
```
import tinycudann as tcnn

for i in range(2):
    torch.cuda.manual_seed(i)
    torch.manual_seed(i)
    mlp = tcnn.NetworkWithInputEncoding(
        n_input_dims=1,
        n_output_dims=1,
        encoding_config={
            "otype": "Identity",
        },
        network_config={
            "otype": "FullyFusedMLP",
            "activation": 'ReLU',
            "output_activation": "None",
            "n_neurons": 16,
            "n_hidden_layers": 1
        }
    )
    print('seed', i, 'mlp', mlp.state_dict()['params'])
```

Output:

```
seed 0 mlp tensor([-0.3051, -0.0251, -0.2823, -0.3212, -0.1474,  0.0245,  0.3514,  0.2916,
         0.1646, -0.3333,  0.1619,  0.0028, -0.3585, -0.3433, -0.0231,  0.2197,
        -0.2476, -0.0576, -0.1039,  0.0664, -0.1978, -0.3768,  0.0883,  0.2568,
        -0.0271,  0.1863, -0.2435, -0.0829,  0.0638, -0.2467,  0.2278,  0.3955,
         0.2545,  0.1406,  0.1448, -0.2271,  0.1772,  0.2663,  0.2252, -0.2054,
         0.2094, -0.1343,  0.0963,  0.3108,  0.3561,  0.0167,  0.3754,  0.4051,
        -0.1719,  0.2596, -0.1522, -0.3503, -0.2521, -0.3742,  0.2345,  0.1655,
         0.2622,  0.3120,  0.0325,  0.1479,  0.2457, -0.0133,  0.2062, -0.4253,
         0.4089,  0.1696, -0.0089,  0.0069,  0.1963,  0.0030, -0.4076,  0.1792,
        -0.3341,  0.2969,  0.2557, -0.3540,  0.2372, -0.0131,  0.3104,  0.3159,
         0.2864, -0.1599,  0.0035, -0.3356, -0.0467, -0.1736,  0.1935, -0.4077,
         0.4295,  0.3938, -0.2081, -0.3999,  0.2518, -0.0681,  0.3530,  0.1220,
         0.1811,  0.0165,  0.0131,  0.1877,  0.2457,  0.1204,  0.0931,  0.1077,
        -0.1620, -0.4176, -0.4172, -0.3863,  0.2984, -0.3147,  0.3615,  0.1867,
         0.1673, -0.4247, -0.2822,  0.1096, -0.0605, -0.1041,  0.0551,  0.2718,
         0.0886, -0.3697,  0.4160,  0.3240, -0.1361,  0.4110,  0.1724,  0.1278,
         0.3079, -0.4114, -0.4318, -0.1751,  0.3546, -0.1004, -0.4083,  0.1626,
        -0.1019,  0.4183,  0.2025,  0.1602,  0.3697, -0.1231,  0.0118,  0.1521,
        -0.1712,  0.1121,  0.2934, -0.4115, -0.2015, -0.0814, -0.2745, -0.2584,
         0.2223,  0.4043, -0.3082,  0.1508, -0.2589,  0.3132, -0.2451, -0.2249,
        -0.1539, -0.1802, -0.1210, -0.4258, -0.0422,  0.2323,  0.0560, -0.3638,
        -0.2910, -0.1884,  0.1325, -0.3853, -0.1530,  0.2587,  0.2357, -0.1157,
        -0.1869,  0.3047,  0.3880,  0.3362,  0.1920, -0.0079,  0.3100, -0.2802,
        -0.3416, -0.0249,  0.1704,  0.3745,  0.0594,  0.4256, -0.2815,  0.2946,
        -0.2979, -0.2557, -0.3768, -0.3274,  0.1992, -0.3114,  0.2294,  0.1209,
        -0.1028, -0.3851,  0.1902,  0.0265,  0.0160,  0.3033,  0.2736, -0.3465,
         0.0468,  0.3018,  0.0984, -0.0882,  0.0595,  0.2953, -0.2128, -0.3089,
        -0.0934,  0.2360,  0.1064,  0.2308, -0.1203, -0.2573, -0.2075, -0.0159,
         0.3798,  0.2399,  0.3748, -0.2451, -0.1408,  0.0666,  0.4195,  0.2497,
        -0.1762, -0.2986,  0.0334,  0.0775,  0.0577, -0.1590, -0.4050,  0.2831,
         0.3559,  0.2021,  0.3678, -0.2423,  0.2809,  0.1262, -0.3110, -0.3399,
         0.2407,  0.0005,  0.3761, -0.3768, -0.2322,  0.1249,  0.2922, -0.0851,
        -0.2197, -0.2368,  0.1406, -0.1251, -0.1667,  0.3343, -0.2141, -0.0556,
         0.2678, -0.4309, -0.2803,  0.2586, -0.1162,  0.3677,  0.0718,  0.2383,
        -0.0267,  0.1713, -0.3114, -0.1192,  0.3777, -0.1000, -0.3825, -0.2357,
         0.1611, -0.3103,  0.0216,  0.2086, -0.0587,  0.4159,  0.1478,  0.3629,
        -0.1017, -0.1424,  0.3363, -0.3827, -0.2672,  0.4147, -0.1659, -0.2847,
        -0.0479,  0.2963,  0.0500,  0.3323, -0.3087,  0.4259, -0.0371, -0.2796,
        -0.1382, -0.2129, -0.0554, -0.2960,  0.1847, -0.3338,  0.1198, -0.0547,
         0.2433,  0.3912,  0.0345, -0.2856, -0.3760,  0.3792, -0.0998,  0.0021,
        -0.0504, -0.3750, -0.0293,  0.3666,  0.4103, -0.3628,  0.2224, -0.0547,
        -0.0210,  0.1796,  0.0924, -0.4309, -0.1750,  0.4132, -0.3847,  0.3794,
         0.2806, -0.1907, -0.3599, -0.2427, -0.2700, -0.4280,  0.3612,  0.4198,
         0.3565, -0.1089,  0.0127,  0.0580,  0.2556, -0.3637, -0.0161,  0.1427,
        -0.2208, -0.0326,  0.1834,  0.3215,  0.1072,  0.2840, -0.2392, -0.3205,
         0.3848,  0.1172, -0.3491, -0.1722,  0.1599, -0.0780,  0.0251,  0.0080,
        -0.2626, -0.0966,  0.3349, -0.4020, -0.1454,  0.0984, -0.2509, -0.1824,
        -0.4154, -0.2858,  0.3266, -0.2316, -0.2937,  0.4046,  0.4069, -0.2952,
         0.1732,  0.3710,  0.2761, -0.3251,  0.1435,  0.2052, -0.4051, -0.3403,
         0.4077,  0.2450, -0.2911, -0.3098, -0.3674,  0.1185,  0.2189, -0.0955,
        -0.0416,  0.3218, -0.4232, -0.2687,  0.0988,  0.4079, -0.0193, -0.2182,
        -0.0433, -0.3884,  0.3574,  0.2511, -0.0715,  0.4036,  0.3800,  0.3846,
         0.1854,  0.1615,  0.4205, -0.0566,  0.0115,  0.2666, -0.1657, -0.3247,
        -0.2862, -0.1423, -0.2736, -0.0573, -0.1693,  0.2782, -0.3895, -0.0174,
         0.2981, -0.4235, -0.4269, -0.3550,  0.3828, -0.0988,  0.4313, -0.3233,
        -0.0770,  0.2166,  0.2867,  0.3963,  0.0404,  0.0349,  0.0709,  0.3580,
        -0.1107, -0.1473, -0.1665,  0.0243, -0.3275, -0.1743, -0.2925, -0.0269,
         0.3800,  0.3799,  0.0195,  0.0359,  0.0543, -0.2167, -0.3081,  0.1839,
         0.3010,  0.1505, -0.0168,  0.4298,  0.2268,  0.3522, -0.2045, -0.4151,
        -0.2676, -0.3250,  0.0243,  0.2265, -0.2434,  0.0730,  0.3245, -0.4228,
        -0.3990,  0.0352,  0.1499, -0.2100,  0.2236,  0.2313, -0.4156,  0.1111,
         0.0401, -0.1046,  0.0487, -0.3760, -0.4254,  0.0931, -0.3287,  0.2265,
         0.3915, -0.2986,  0.1570,  0.2463,  0.2019,  0.2405,  0.0022, -0.0725,
         0.0488, -0.2816, -0.3866,  0.3281,  0.0436,  0.2818,  0.2785,  0.3578],
       device='cuda:0')
seed 1 mlp tensor([-0.3051, -0.0251, -0.2823, -0.3212, -0.1474,  0.0245,  0.3514,  0.2916,
         0.1646, -0.3333,  0.1619,  0.0028, -0.3585, -0.3433, -0.0231,  0.2197,
        -0.2476, -0.0576, -0.1039,  0.0664, -0.1978, -0.3768,  0.0883,  0.2568,
        -0.0271,  0.1863, -0.2435, -0.0829,  0.0638, -0.2467,  0.2278,  0.3955,
         0.2545,  0.1406,  0.1448, -0.2271,  0.1772,  0.2663,  0.2252, -0.2054,
         0.2094, -0.1343,  0.0963,  0.3108,  0.3561,  0.0167,  0.3754,  0.4051,
        -0.1719,  0.2596, -0.1522, -0.3503, -0.2521, -0.3742,  0.2345,  0.1655,
         0.2622,  0.3120,  0.0325,  0.1479,  0.2457, -0.0133,  0.2062, -0.4253,
         0.4089,  0.1696, -0.0089,  0.0069,  0.1963,  0.0030, -0.4076,  0.1792,
        -0.3341,  0.2969,  0.2557, -0.3540,  0.2372, -0.0131,  0.3104,  0.3159,
         0.2864, -0.1599,  0.0035, -0.3356, -0.0467, -0.1736,  0.1935, -0.4077,
         0.4295,  0.3938, -0.2081, -0.3999,  0.2518, -0.0681,  0.3530,  0.1220,
         0.1811,  0.0165,  0.0131,  0.1877,  0.2457,  0.1204,  0.0931,  0.1077,
        -0.1620, -0.4176, -0.4172, -0.3863,  0.2984, -0.3147,  0.3615,  0.1867,
         0.1673, -0.4247, -0.2822,  0.1096, -0.0605, -0.1041,  0.0551,  0.2718,
         0.0886, -0.3697,  0.4160,  0.3240, -0.1361,  0.4110,  0.1724,  0.1278,
         0.3079, -0.4114, -0.4318, -0.1751,  0.3546, -0.1004, -0.4083,  0.1626,
        -0.1019,  0.4183,  0.2025,  0.1602,  0.3697, -0.1231,  0.0118,  0.1521,
        -0.1712,  0.1121,  0.2934, -0.4115, -0.2015, -0.0814, -0.2745, -0.2584,
         0.2223,  0.4043, -0.3082,  0.1508, -0.2589,  0.3132, -0.2451, -0.2249,
        -0.1539, -0.1802, -0.1210, -0.4258, -0.0422,  0.2323,  0.0560, -0.3638,
        -0.2910, -0.1884,  0.1325, -0.3853, -0.1530,  0.2587,  0.2357, -0.1157,
        -0.1869,  0.3047,  0.3880,  0.3362,  0.1920, -0.0079,  0.3100, -0.2802,
        -0.3416, -0.0249,  0.1704,  0.3745,  0.0594,  0.4256, -0.2815,  0.2946,
        -0.2979, -0.2557, -0.3768, -0.3274,  0.1992, -0.3114,  0.2294,  0.1209,
        -0.1028, -0.3851,  0.1902,  0.0265,  0.0160,  0.3033,  0.2736, -0.3465,
         0.0468,  0.3018,  0.0984, -0.0882,  0.0595,  0.2953, -0.2128, -0.3089,
        -0.0934,  0.2360,  0.1064,  0.2308, -0.1203, -0.2573, -0.2075, -0.0159,
         0.3798,  0.2399,  0.3748, -0.2451, -0.1408,  0.0666,  0.4195,  0.2497,
        -0.1762, -0.2986,  0.0334,  0.0775,  0.0577, -0.1590, -0.4050,  0.2831,
         0.3559,  0.2021,  0.3678, -0.2423,  0.2809,  0.1262, -0.3110, -0.3399,
         0.2407,  0.0005,  0.3761, -0.3768, -0.2322,  0.1249,  0.2922, -0.0851,
        -0.2197, -0.2368,  0.1406, -0.1251, -0.1667,  0.3343, -0.2141, -0.0556,
         0.2678, -0.4309, -0.2803,  0.2586, -0.1162,  0.3677,  0.0718,  0.2383,
        -0.0267,  0.1713, -0.3114, -0.1192,  0.3777, -0.1000, -0.3825, -0.2357,
         0.1611, -0.3103,  0.0216,  0.2086, -0.0587,  0.4159,  0.1478,  0.3629,
        -0.1017, -0.1424,  0.3363, -0.3827, -0.2672,  0.4147, -0.1659, -0.2847,
        -0.0479,  0.2963,  0.0500,  0.3323, -0.3087,  0.4259, -0.0371, -0.2796,
        -0.1382, -0.2129, -0.0554, -0.2960,  0.1847, -0.3338,  0.1198, -0.0547,
         0.2433,  0.3912,  0.0345, -0.2856, -0.3760,  0.3792, -0.0998,  0.0021,
        -0.0504, -0.3750, -0.0293,  0.3666,  0.4103, -0.3628,  0.2224, -0.0547,
        -0.0210,  0.1796,  0.0924, -0.4309, -0.1750,  0.4132, -0.3847,  0.3794,
         0.2806, -0.1907, -0.3599, -0.2427, -0.2700, -0.4280,  0.3612,  0.4198,
         0.3565, -0.1089,  0.0127,  0.0580,  0.2556, -0.3637, -0.0161,  0.1427,
        -0.2208, -0.0326,  0.1834,  0.3215,  0.1072,  0.2840, -0.2392, -0.3205,
         0.3848,  0.1172, -0.3491, -0.1722,  0.1599, -0.0780,  0.0251,  0.0080,
        -0.2626, -0.0966,  0.3349, -0.4020, -0.1454,  0.0984, -0.2509, -0.1824,
        -0.4154, -0.2858,  0.3266, -0.2316, -0.2937,  0.4046,  0.4069, -0.2952,
         0.1732,  0.3710,  0.2761, -0.3251,  0.1435,  0.2052, -0.4051, -0.3403,
         0.4077,  0.2450, -0.2911, -0.3098, -0.3674,  0.1185,  0.2189, -0.0955,
        -0.0416,  0.3218, -0.4232, -0.2687,  0.0988,  0.4079, -0.0193, -0.2182,
        -0.0433, -0.3884,  0.3574,  0.2511, -0.0715,  0.4036,  0.3800,  0.3846,
         0.1854,  0.1615,  0.4205, -0.0566,  0.0115,  0.2666, -0.1657, -0.3247,
        -0.2862, -0.1423, -0.2736, -0.0573, -0.1693,  0.2782, -0.3895, -0.0174,
         0.2981, -0.4235, -0.4269, -0.3550,  0.3828, -0.0988,  0.4313, -0.3233,
        -0.0770,  0.2166,  0.2867,  0.3963,  0.0404,  0.0349,  0.0709,  0.3580,
        -0.1107, -0.1473, -0.1665,  0.0243, -0.3275, -0.1743, -0.2925, -0.0269,
         0.3800,  0.3799,  0.0195,  0.0359,  0.0543, -0.2167, -0.3081,  0.1839,
         0.3010,  0.1505, -0.0168,  0.4298,  0.2268,  0.3522, -0.2045, -0.4151,
        -0.2676, -0.3250,  0.0243,  0.2265, -0.2434,  0.0730,  0.3245, -0.4228,
        -0.3990,  0.0352,  0.1499, -0.2100,  0.2236,  0.2313, -0.4156,  0.1111,
         0.0401, -0.1046,  0.0487, -0.3760, -0.4254,  0.0931, -0.3287,  0.2265,
         0.3915, -0.2986,  0.1570,  0.2463,  0.2019,  0.2405,  0.0022, -0.0725,
         0.0488, -0.2816, -0.3866,  0.3281,  0.0436,  0.2818,  0.2785,  0.3578],
       device='cuda:0')```

**Comment by Tom94** [comment]
Hi, you can pass your seed to tcnn as `tcnn.NetworkWithInputEncoding(..., seed=i)`.

I'm not sure why I haven't used pytorch's native seed system back when implementing the tcnn bindings but I'm hesitant to change it now for backward compatibility reasons -- sorry about that.
